drivers/tee/optee/optee.c

```diff
@@ -1230,6 +1268,16 @@ static int optee_init(const struct device *dev)
 		return -ENOTSUP;
 	}
 
+	/*
+	 * We need to disable all existing shm objects before enabling caching.
```

Comment:
Could you please elaborate this? How it is possible that OP-TEE has cached shm buffers before driver is even being initialized?

Reply:
So, the question still stands. 

If this is a leftover from, say, U-Boot, then U-Boot should be fixed.

Reply:
According to the comment from https://elixir.bootlin.com/linux/latest/source/drivers/tee/optee/smc_abi.c#L1749
There is a posibility that optee still has shm registered and no properly cleaned from the upper layer. I think this is not the case if zephyr is running as XEN domain, but I assume that there is a case when for example we start zephyr from u-boot after accessing optee from u-boot or other entity, such as older kernels, where resources are not properly cleaned.
I aware that it's unlikely to happen but this is some level of protection which is nice to have